types/field.go

```diff
@@ -0,0 +1,88 @@
+package types
+
+import (
+	"reflect"
+)
+
+type TagSet map[string]struct{}
+
+func newTagSet(tags ...string) TagSet {
+	set := make(TagSet, len(tags))
+	for _, tag := range tags {
+		set[tag] = struct{}{}
+	}
+	return set
+}
+
+func (t TagSet) Contains(tag string) bool {
+	_, ok := t[tag]
+	return ok
+}
+
+type Field struct {
+	Tags  TagSet
+	value reflect.Value
+	Type  reflect.Type
+	Kind  reflect.Kind
+	Ptr   bool
+	IsNil bool
+}
+
+func FieldFromValue(in interface{}, tags []string) *Field {
```

Comment:
ok, having read the whole PR now, I think I'd like to see this in `_test.go` only - it doesn't seem like there is a reason to use this outside of test.

Reply:
I've removed this entirely